python/pyspark/sql/tests/test_pandas_udf_scalar.py

```diff
@@ -134,6 +134,24 @@ def test_pandas_udf_nested_arrays(self):
         result = df.select(tokenize("vals").alias("hi"))
         self.assertEqual([Row(hi=[["hi", "boo"]]), Row(hi=[["bye", "boo"]])], result.collect())
 
+    def test_pandas_array_struct(self):
+        @pandas_udf("Array<struct<col1:string, col2:long, col3:double>>")
+        def return_cols(cols):
+            return cols
```

Comment:
Can we add an assert on its data type here too?

Reply:
You can refer to https://github.com/apache/spark/pull/33876 about where to add the tests.